src/Zip/ZipFile.Selector.cs

```diff
@@ -1255,8 +1255,8 @@ internal partial class NameCriterion : SelectionCriterion
     {
         internal override bool Evaluate(Ionic.Zip.ZipEntry entry)
         {
-            // swap forward slashes in the entry.FileName for backslashes
-            string transformedFileName = entry.FileName.Replace("/", "\\");
+            // swap slashes in reference to local configuration
+            string transformedFileName = entry.FileName.Replace(Path.DirectorySeparatorChar == '/' ? '\\' : '/', Path.DirectorySeparatorChar);

```

Comment:
What do you mean? This just makes sure that the file name have the platform specific seperator, else the later used GetFilename function will fail